apps/aecore/src/aec_chain_state.erl

```diff
@@ -267,17 +267,22 @@ get_hash_at_height(Height, State) when is_integer(Height), Height >= 0 ->
                         error -> error({broken_chain, Height});
                         {ok, [Node]} -> {ok, hash(Node)};
                         {ok, [_|_] = Nodes} ->
-                            first_hash_in_main_chain(Nodes, Hash)
+                            keyblock_hash_in_main_chain(Nodes, Hash)
                     end
             end
     end.
 
-first_hash_in_main_chain([Node|Left], TopHash) ->
-    case hash_is_in_main_chain(hash(Node), TopHash) of
-        true  -> {ok, hash(Node)};
-        false -> first_hash_in_main_chain(Left, TopHash)
+keyblock_hash_in_main_chain([Node|Left], TopHash) ->
+    NodeFound =
+        case hash_is_in_main_chain(hash(Node), TopHash) of
```

Comment:
Change order of these. hash_is_in_main_chain is much more expensive.

Reply:
Also, leave a `TODO` that this can change when key blocks and micro blocks are indexed separately.